earth_enterprise/src/fusion/autoingest/sysman/khResourceManager.cpp

```diff
@@ -262,11 +268,14 @@ khResourceManager::EraseProvider(khResourceProviderProxy *proxy)
   // remove the Volumes that this provider used to manage
   std::vector<std::string> volnames;
   GetHostVolumes(host, volnames);
-  for (std::vector<std::string>::const_iterator vn = volnames.begin();
-       vn != volnames.end(); ++vn) {
-    delete volumes[*vn];
-    volumes.erase(*vn);
-  }
+  //for (std::vector<std::string>::const_iterator vn = volnames.begin();
+  //     vn != volnames.end(); ++vn) {
+  for (const auto& vn : volnames) {
+    delete volumes[vn];
+    volumes.erase(vn);
+  }
+  if (volumes.empty())
+      volumes.clear();
```

Comment:
Minor: I like putting brackets around all if statements. The only time I make an exception is when I put the entire statement on one line: `if (volumes.empty()) volumes.clear()`

Reply:
Will do